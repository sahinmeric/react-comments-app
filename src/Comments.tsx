import React from "react";
import axios from "axios";
import { useQuery } from "@tanstack/react-query";

import NotFound from "./NotFound";
import ShowProgress from "./ShowProgress";
import CommentCard from "./CommentCard"
import { Typography } from "@mui/material";


export type CommentType = {
    commentId: number;
    id: number;
    name: string;
    email: string;
    body: string;
};

export function useComments() {
    return useQuery(["comments"], async (): Promise<Array<CommentType>> => {
        const { data } = await axios.get(
            "http://jsonplaceholder.typicode.com/posts/1/comments"
        );
        return data;
    });
}

function Comments() {
    const { data: comment, error, isFetching, isLoading } = useComments();
    if (isLoading || isFetching) {
        return <ShowProgress />
    }
    if (error || !comment) {
        return <NotFound />
    }
    return (
        <>
            <ul>
                <Typography
                    variant="h3"
                    style={{
                        padding: 10,
                        margin: "auto",
                        width: "50%",
                        textAlign: "center",
                        color:"#192F52"
                    }}>
                    Comments
                </Typography>
                {comment?.map((comment) => (
                    <li key={comment.id}>
                        <CommentCard
                            commentId={comment.commentId}
                            id={comment.id}
                            name={comment.name}
                            email={comment.email}
                            body={comment.body}
                        />
                    </li>
                ))}
            </ul>
        </>
    );
}

export default Comments;
