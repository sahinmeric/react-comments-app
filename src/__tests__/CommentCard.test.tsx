/**
 * @jest-environment jsdom
 */
import { describe, expect, test } from '@jest/globals';
import { queryByTestId, render, screen } from '@testing-library/react';
import ReactDOM from 'react-dom';
import Comments from '../Comments';
import CommentCard from '../CommentCard';
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";


const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            cacheTime: 1000 * 60 * 60 * 24,
            refetchOnWindowFocus: false,
            refetchOnMount: false,
        },
    },
});

describe('CommentCard test suit', () => {
    beforeAll(() => {
        /* Runs before all tests */
    })
    afterAll(() => {
        /* Runs after all tests */
    })
    beforeEach(() => {
        /* Runs before each test */
    })
    afterEach(() => {
        /* Runs after each test */
    })

    test('if it renders without crash', () => {
        const component = renderer.create(
            <BrowserRouter>
                <QueryClientProvider client={queryClient}>
                    <CommentCard commentId={1}
                        id={2}
                        name={"asdasda"}
                        email={"email"}
                        body={"this is comentario"} />
                </QueryClientProvider>
            </BrowserRouter>)
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});

