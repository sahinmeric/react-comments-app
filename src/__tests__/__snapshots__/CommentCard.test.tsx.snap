// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard test suit if it renders without crash 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  style={
    {
      "margin": "auto",
      "padding": "2px",
    }
  }
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <a
      href="/2"
      onClick={[Function]}
      style={
        {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 css-1qz17ui-MuiPaper-root"
        style={
          {
            "backgroundColor": "#C6EBEB",
            "border": "1px solid lightgray",
            "padding": "10px 10px",
          }
        }
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                alt="Avatar"
                className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="test-file-stub"
              />
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-true css-114cq88-MuiGrid-root"
          >
            <h4
              style={
                {
                  "color": "#192F52",
                  "margin": "5px",
                  "padding": "5px 0",
                  "textAlign": "left",
                }
              }
            >
              <b>
                asdasda
              </b>
            </h4>
            <em
              style={
                {
                  "color": "#192F52",
                  "padding": "5px",
                }
              }
            >
              “
              this is comentario
              ”
            </em>
            <p
              style={
                {
                  "color": "gray",
                  "textAlign": "right",
                }
              }
            >
              email
            </p>
          </div>
        </div>
      </div>
    </a>
    <span
      className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
